### ISSUE TITLE
Gradle plugin: properly pass generated logback.xml to gatlingRun tasks

### ISSUE THREAD

**Issue by eshepelyuk** [gatling/gatling#3948]
### Scope of changes

* In case `logback.xml` is generated by plugin used - the file should be passed to every forked JVM that that executes Gatling via additional `-Dlogback.configurationFile` system property as [logback docs](http://logback.qos.ch/manual/configuration.html#configFileProperty) suggests.
* currently it is just generated into hardcoded location within `build` directory, that may bring classpath errors and clashes.
* Make sure all existing tests work, at least file location should be changed.

Blocked by #3952

**Comment by slandelle** [comment]
I agree we should use the `SourceSet` API, but we shouldn't search in the `resources` directory by in `output` one. Ultimately, what we want is to detect files that are in the classpath, not files that will be/have been copied somewhere in the classpath.

**Comment by slandelle** [comment]
> Another issue is that existing logback detection is bound to hardcoded path (src/gatling/resources) and neglects files other than logback.xml

This is a bug that should have its own ticket, not part a this refactoring. Agree?

**Comment by slandelle** [comment]
I extracted the "other issue" part as a dedicated ticket as it's not strictly related to this refactoring, see #3952

**Comment by eshepelyuk** [comment]
oh my god, why :((

what you had extracted is exaclty this issue, they can not be separated.
please close #3952, it's excessive

**Comment by slandelle** [comment]
Why would missing to detect `logback-test.xml` and `logback.groovy` files be related to the large refactoring with is the topic of this issue?
#3952 could perfectly be implemented on top of the existing implementation without this refactoring.

PS: please refrain from using disrespectful comments such as "oh my god, why :((".
